library/lib/core/application.cpp

```diff
@@ -939,7 +940,12 @@ std::string Application::getLocale()
 void Application::addToFreeQueue(View* view)
 {
     brls::Logger::verbose("Application::addToFreeQueue {}", view->describe());
-    deletionPool.insert(view);
+    
+    for (brls::View* viewToDelete : Application::deletionPool)
```

Comment:
I think could be better to use

```
if (std::binary_search(Application::deletionPool.begin(), Application::deletionPool.end(), view))
        return;
```

Also put Logger after if check, in case we skipped this view to add to queue, I don't think it should log